### PR TITLE
Expose the rest of the actions to the node bindings

### LINKED ISSUE #1393: Wrap the rest of the actions for the node bindings
#1341 created the node bindings, but only wrapped the `route` action. We should get the rest of the actions in there too so that the bindings can be considered complete. Once we wrap them all (and confirm binary portability), we should publish the module to the npm registry. 

- bindings & tests for: 
   - [ ] match
   - [ ] matrix
   - [ ] locate
   - [ ] optimized_route
   - [ ] isochrone
   - [ ] trace_route
   - [ ] trace_attributes
   - [ ] height
   - [ ] transit_available

### PR DESCRIPTION
**Pull request by allierowan**
# Issue

this PR exposes the other tyr actions to the bindings.
   - [x] matrix
   - [x] locate
   - [x] optimized_route
   - [x] isochrone
   - [x] trace_route
   - [x] trace_attributes
   - [x] height
   - [x] transit_available

closes #1393 

## Tasklist

 - [x] Add tests
 - [ ] Review - you must request approval to merge any PR to master
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)
 - [x] Update relevant [documentation](docs/README.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/bindings/node/node.cc [hunk 1]
```diff
@@ -213,6 +221,158 @@ class Actor {
     return outStr;
   }
 
+  static napi_value Locate(napi_env env, napi_callback_info info) {
```
Comment (danpaz) [inline src/bindings/node/node.cc]: There's quite a bit of repetition in each of these wrapper functions. Did you look into whether we could take advantage of [macros](http://www.cplusplus.com/doc/tutorial/preprocessor/) or some other construct to reduce repeated code?

Comment (kevinkreiser) [reply src/bindings/node/node.cc]: i think you can do this pretty easily with lambdas and `std::function` (which is in `#include <functional>`):

```
  static napi_value Locate(napi_env env, napi_callback_info info) {
    return generic_action(env, info, [](actor_t* actor, const std::string& request) -> std::string {
        return actor->locate(request);
      }); // <---- this feels pretty javascripty amiright!?
  }

  //TODO: all the other functions should look basically just like this ^^

  static napi_value generic_action(napi_env env, napi_callback_info info, 
      const std::function<std::string(actor_t*, const std::string&)>& func) {
    napi_value jsthis;
    napi_status status;

    std::string reqString = ParseRequest(env, info, &jsthis);

    Actor* obj;
    status = napi_unwrap(env, jsthis, reinterpret_cast<void**>(&obj));
    checkNapiStatus(status, env, "Failed to unwrap js object");

    std::string locate_json;
    try {
      locate_json = func(actor, reqString);
    } catch (const std::exception& e) {napi_throw_error(env, NULL, e.what()); }

    auto outStr = WrapString(env, locate_json);
    return outStr;
  }
```

Comment (allierowan) [reply src/bindings/node/node.cc]: gave it a shot with a few edits to satisfy the compiler. hopefully it works!

File: test/bindings/nodejs/transit_available.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  var badRequest = '{"locations": [{"lon": 500, "lat": 40.543564, "radius": 20},{"lon": 500, "lat": 40.543939, "radius": 20}]}';
+  var transitAvailable = JSON.parse(valhalla.transitAvailable(badRequest));
+  assert.ok(transitAvailable);
+  assert.equal(transitAvailable[0].istransit, false, 'no transit found');
+  assert.end();
+});
+
+test('transitAvailable: returns an error if request format is wrong', function(assert) {
+  var badRequest = '{"locations":[40.546115,-76.385076], [40.544232,"lon":-76.385752],"costing":"auto"}';
+  // TODO: we are not throwing a very useful error in this case - we should track this
```
Comment (danpaz) [inline test/bindings/nodejs/transit_available.test.js]: It's great we have these tests to document the behavior, but these inline TODOs are not actionable. I think we should remove the comments and instead write a ticket to follow up on throwing better errors.

File: test/bindings/nodejs/transit_available.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  // down and throw something a little more descriptive
+  assert.throws(() => { valhalla.transitAvailable(badRequest) }, /std::exception/, 'Throws an error when request format is wrong');
+  assert.end();
+});
+
+test.onFinish(() => {
+    valhalla = null;
+    if (global.gc) global.gc();
+});
```
Comment (danpaz) [inline test/bindings/nodejs/transit_available.test.js]: I think we can remove these `onFinish`es and calls to gc. If this causes tests to fail we should probably dig a little further.

File: package.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-  "version": "3.0.0-alpha.1",
+  "version": "3.0.0-rc.1",
   "main": "lib/index.js",
   "directories": {
     "doc": "docs",
     "test": "test"
   },
   "scripts": {
-    "test": "node --expose-gc test/nodejs/*.test.js",
+    "test": "tape test/bindings/nodejs/*.test.js --expose-gc",
```
Comment (danpaz) [inline package.json]: If we remove calls to gc we can likely remove this `--expose-gc` flag as well.

File: .circleci/config.yml [hunk 5]
```diff
@@ -48,6 +48,8 @@ jobs:
           command: |
             export PACKAGE_JSON_VERSION=$(node -e "console.log(require('./package.json').version)")
             export PUBLISH=$([[ "${CIRCLE_TAG:-}" == "v${PACKAGE_JSON_VERSION}" ]] && echo "On" || echo "Off")
+            echo ${CIRCLE_TAG:-}
+            echo ${PACKAGE_JSON_VERSION}
```
Comment (danpaz) [inline .circleci/config.yml]: Should we add `npm test` to these CI builds?

Comment (danpaz) [reply .circleci/config.yml]: This is addressed in https://github.com/valhalla/valhalla/pull/1415/files#diff-15547c54d3d4898a882b4ab7b3cee381R148, ignore.

File: test/bindings/nodejs/transit_available.test.js (hunk 2 above)
Comment (allierowan) [reply test/bindings/nodejs/transit_available.test.js]: there is already a ticket - a member of the community mentioned a similar concern. I'll remove the TODOs, although I have noticed that this is a practice that is common in this codebase.

Comment (kevinkreiser) [reply test/bindings/nodejs/transit_available.test.js]: imho these `TODO:`s aren't harmful. maybe a more appropriate label would be `NOTE:` to remove the idea that you have to do something at this point in the code. to me it reads like a note about expectations at this point. just my 2 cents, feel free to ignore

**Pull request merged**